things/service.go

```diff
@@ -147,6 +147,22 @@ type thingsService struct {
 	ulidProvider mainflux.IDProvider
 }
 
+// createUID: take the external id if supplied, otherwise use default provider
+// validate the external id, if validation fails, generate an id using the
+// default provider.
+func (ts *thingsService) createUID(extid string) (id string, err error) {
+	if(len(extid) > 0) {
```

Comment:
Avoid if/else and don't ignore errors. Also, I don't think that is good to generate an UUID if exterID is wrong. Better return an error. Something like: 
```
	if extID == "" {
		id, err = ts.idProvider.ID()
		if err != nil {
			return "", err
		}

		return id, nil
	}

	id, err = ts.idProvider.FromString(extID)
	if err != nil {
		return "", err
	}
	return id, nil
```

Reply:
I have a few comments about it, will put in  a separate comment in some time. :-)